vector_setters.go

```diff
@@ -32,108 +33,319 @@ func setPrimitive[T any](vec *vector, rowIdx C.idx_t, v T) {
 	xs[rowIdx] = v
 }
 
-func (vec *vector) setTS(t Type, rowIdx C.idx_t, val any) {
-	v := val.(time.Time)
+// TODO: support Decimal here and bool and hugeint
+func setNumeric[S any, T numericType](vec *vector, rowIdx C.idx_t, val S) error {
+	var fv T
+	switch v := any(val).(type) {
+	case uint8:
+		fv = T(v)
+	case int8:
+		fv = T(v)
+	case uint16:
+		fv = T(v)
+	case int16:
+		fv = T(v)
+	case uint32:
+		fv = T(v)
+	case int32:
+		fv = T(v)
+	case uint64:
+		fv = T(v)
+	case int64:
+		fv = T(v)
+	case uint:
+		fv = T(v)
+	case int:
+		fv = T(v)
+	case float32:
+		fv = T(v)
+	case float64:
+		fv = T(v)
+	case bool:
+		if v {
+			fv = 1
+		} else {
+			fv = 0
+		}
+	case *big.Int:
+		if v == nil {
+			return castError(reflect.TypeOf(val).String(), reflect.TypeOf(fv).String())
+		}
+		if v.IsUint64() {
+			fv = T(v.Uint64())
+		} else {
+			fv = T(v.Int64())
+		}
+	case Decimal:
+		if v.Value == nil {
+			return castError(reflect.TypeOf(val).String(), reflect.TypeOf(fv).String())
+		}
+		if v.Value.IsUint64() {
+			fv = T(v.Value.Uint64())
+		} else {
+			fv = T(v.Value.Int64())
+		}
+	default:
+		return castError(reflect.TypeOf(val).String(), reflect.TypeOf(fv).String())
+	}
+	setPrimitive(vec, rowIdx, fv)
+	return nil
+}
+
+func setBool[S any](vec *vector, rowIdx C.idx_t, val S) error {
```

Comment:
I don't understand why we need so many cases here. Shouldn't the following be sufficient?
```Go
func setBool[S any](vec *vector, rowIdx C.idx_t, val S) error {
	var fv bool
	switch v := any(val).(type) {
	case bool:
		fv = v
	default:
		return castError(reflect.TypeOf(val).String(), reflect.TypeOf(fv).String())
	}
	setPrimitive(vec, rowIdx, fv)
	return nil
}
```

Reply:
copy-paste from above: Sure, I mostly added all the cases because I thought it would be nice to support casting from as many types as possible, but I can reduce the amount of cases if you want. Should I just reduce the supported types to what was possible before?